### PR TITLE
[DRAFT] ELEMENTS-1309: add retention's governance mode ability

### PR DESCRIPTION
**Pull request by RSalem07**
PR related to the governance mode ability will require the back end one https://github.com/nuxeo/nuxeo/pull/4594

**TO NOT MERGE**

### CODE REVIEW

**Review by andreacornaglia — APPROVED** [review]
LGTM

File: ui/nuxeo-filters-behavior.js [hunk 1]
```diff
@@ -273,4 +273,23 @@ export const FiltersBehavior = {
...
+    if (!underRetentionOrLegalHold && hasRemovePermission) {
+      return true;
+    }
+    return (
+      (this.hasAdministrationPermissions(user) || this.hasPermission(document, 'RemoveRecords')) &&
+      Nuxeo &&
+      Nuxeo.UI &&
+      Nuxeo.UI.config &&
+      Nuxeo.UI.config.retention &&
+      String(Nuxeo.UI.config.retention.mode) === 'governance'
```
Comment (nelsonsilva) [inline ui/nuxeo-filters-behavior.js:292]: so if `Nuxeo.UI.config.retention.mode` is not set we can't delete documents ?
also, shouldn't user permissions on the document be enough to do this checks? the ideal of these hardcoded client side checks is for them to mimic what the server allows you to do, if toggling a client side flag prevents me from deleting a document but the REST API still allows it then this is a false sense of security so it should go in a filter in a UI contribution at most.

Comment (RSalem07) [reply ui/nuxeo-filters-behavior.js:292]: Hey @nelsonsilva thanks for the feedbacks, I was going to ask you if we can meet. I want to share with you these kind of needs/requirements. I have opened a PR on backend part for almost the same rules. on frontend we want to implement the same things to hide or show the button(s). I asked myself how we can improve these double check, how we can do it easily; contribution, inheritance of elements ... I'll to send an invitation this week if you agree to talk about it.

File: ui/nuxeo-filters-behavior.js [hunk 2]
```diff
@@ -273,4 +273,23 @@ export const FiltersBehavior = {
     const processes = document.contextParameters.runnableWorkflows;
     return processes && processes.length > 0;
   },
+
+  /**
+   * Checks if the given document or its main content, can be deleted by the given user
```
Comment (Gabez0r) [inline ui/nuxeo-filters-behavior.js:278]:
```suggestion
   * Checks if the given document or its main content, can be deleted by the given user.
```

Comment (Gabez0r) [reply ui/nuxeo-filters-behavior.js:278]: WDYM by "or its main content"?

File: ui/nuxeo-filters-behavior.js [hunk 3]
```diff
@@ -273,4 +273,23 @@ export const FiltersBehavior = {
...
+   */
+  canDelete(document, user) {
+    const underRetentionOrLegalHold = this.isUnderRetentionOrLegalHold(document);
+    const hasRemovePermission = this.hasPermission(document, 'Remove');
+    if (!underRetentionOrLegalHold && hasRemovePermission) {
+      return true;
+    }
+    return (
+      (this.hasAdministrationPermissions(user) || this.hasPermission(document, 'RemoveRecords')) &&
```
Comment (Gabez0r) [inline ui/nuxeo-filters-behavior.js:287]: Shouldn't we check for the 'Remove' permission here as well? Or does the RemoveRecords permission already includes the 'Remove' one? Where can I find the permission's declaration, btw?

**Pull request closed without merging**